rfc/rfc017-distributed-network-grpc-v2.md

```diff
@@ -54,26 +54,116 @@ As specified by RFC015, the node MUST authenticate the peer's node DID as follow
+Although the set reconciliation protocol can also synchronize any difference in transactions sets between nodes, it's not very efficient for small differences.
 
-#### 4.1.1 XOR
+### 5.1 Data requirements
+
+#### 5.1.1 XOR
 
 The exclusive-or (XOR) value is calculated over all transaction references within a range.
 The size of the XOR value is therefore the same as the transaction reference size (SHA256: 32 bytes).
 The XOR value is used to quickly determine if two nodes have processed the exact same set of transactions.
 
-#### 4.1.2 Invertible Bloom Lookup Table
+### 5.2 Operation
+
+All messages and values mentioned in this chapter are scoped to a single connection between two nodes.
+*Alice* and *Bob* are used to represent two nodes connected to each other.
+
+The protocol generally operates as follows:
+
+1. Alice sends at a set interval (§5.2.1):
+   * XOR of all known transaction references.
+   * The highest LC value.
+   * A list of transaction references since the previous message. If no previous message has been send, the list is empty.
+
+2. When receiving Alice's message, Bob compares the XOR value from Alice with its own (§5.2.2):
+   * When the XOR is the same: no action required.
+   * When different:
+      * Remove all known transaction references from the list.
+      * Request the remaining transactions.
+      * If the remaining list is empty and the LC value is equal or higher than Bob's highest LC value, Bob sends a `State` message as defined in Chapter 6. 
+
+3. Alice responds with the requested transactions (§5.2.3).
+
+4. After adding Alice's transactions to the DAG (making sure its cryptographic signature is valid), Bob SHOULD query the payload if it's missing.
+
+A node MUST make sure to only add transactions of which all previous transactions are present. If previous transactions are missing, Bob will send a `State` message.
+
+#### 5.2.1 Broadcasting new transactions
+
+A node's new transaction references MUST be broadcast at an interval using the `Gossip` message, by default every 2 seconds. The interval MAY be adjusted by the node operator but MUST conform to the limits (min/max interval) defined by the network. It is advised to keep it relatively short because it directly influences the speed by which new transactions are retrieved.
+
+The `Gossip` message MUST contain an `XOR` value, an `LC` value and a list of transaction references (`transactions`).
+The list MUST contain all new transaction references since the last `Gossip` message.
+The list of transaction references MUST be tracked per connection.
+If no new transactions have been added/received or for the first message for a connection, an empty list is sent.
+The list MUST not contain more than 100 transactions.
+This could create a backlog of messages at the sending node's side. 
+A node SHOULD take precautions on keeping this backlog to a minimum, eg it SHOULD reduce the inflow of new transactions.
```

Comment:
If the purpose of reducing the inflow is to keep the gossip protocol working, I'd say the functioning of the gossip protocol is less important than the inflow of transactions. So let the transactions come, but either skip some for the gossip protocol (let it be handled by the IBLT) or just keep sending them while respecting the limits of the network. You'll get 100 TXs every 2 seconds, so 3000 per minute. That's a lot of new TXs.

Yes, the creation of new TXs should be limited/throttled and the TX format should allow for batches for those cases its justified (e.g. new vendor publishing many, many private VCs).

Reply:
I'll rephrase it to keep the backlog below maximum. Because with more than 100TX per 2 seconds, it'll switch to a slower mechanism which will perform even less!